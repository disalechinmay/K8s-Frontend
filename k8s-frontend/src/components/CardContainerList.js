import React, { Component } from "react";
import { ListGroup } from "react-bootstrap";

class CardContainerList extends Component {
	render() {
		return (
			<React.Fragment>
				<ListGroup variant="flush">
					{this.props.list.map((val, index) => {
						return (
							<ListGroup.Item key={index + "_LG_ITEM"}>
								<span className="text-muted">
									Container #{index + 1}:
								</span>{" "}
								{val}
							</ListGroup.Item>
						);
					})}
				</ListGroup>
			</React.Fragment>
		);
	}
}

export default CardContainerList;
