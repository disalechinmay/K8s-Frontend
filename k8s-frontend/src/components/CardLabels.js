import React, { Component } from "react";
import { Button, Badge } from "react-bootstrap";

class CardLabels extends Component {
	render() {
		return (
			<React.Fragment>
				{this.props.labels &&
					Object.entries(this.props.labels).map(
						([key, val], index) => {
							return (
								<React.Fragment key={index + "_LABEL_FRAG"}>
									<Button
										variant={
											(this.props.matchedLabels &&
												this.props.matchedLabels.hasOwnProperty(
													key
												) &&
												val ===
													this.props.matchedLabels[
														key
													] &&
												"success") ||
											"secondary"
										}
										size="sm"
										className="mb-1 text-wrap"
										key={index + "_LABEL_BUTTON"}
									>
										{key}{" "}
										<Badge
											variant="light"
											key={index + "_LABEL_BADGE"}
											className="text-wrap"
										>
											{val}
										</Badge>
									</Button>
									&nbsp;&nbsp;
								</React.Fragment>
							);
						}
					)}
			</React.Fragment>
		);
	}
}

export default CardLabels;
