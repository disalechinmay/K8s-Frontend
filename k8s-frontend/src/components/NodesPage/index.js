export { default as NodesPage } from "./NodesPage";
export { default as NodeCard } from "./NodeCard";
