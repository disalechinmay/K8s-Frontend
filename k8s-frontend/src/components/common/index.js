export { default as CardContainerList } from "./CardContainerList";
export { default as CardLabels } from "./CardLabels";
export { default as LoadingPage } from "./LoadingPage";
export { default as SmallLoadingPage } from "./SmallLoadingPage";
export { default as ErrorPage } from "./ErrorPage";
export { default as SmallErrorPage } from "./SmallErrorPage";
