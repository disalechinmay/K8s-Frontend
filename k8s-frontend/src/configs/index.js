export { API_LOCATION } from "./api";
export { SIDEBAR_OPTIONS, PAGE_LOAD_LIMIT_TIME } from "./constants";
