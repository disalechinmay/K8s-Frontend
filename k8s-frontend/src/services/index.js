export { getNamespaces } from "./namespaces";
export { getNodes } from "./nodes";
export { getPods } from "./pods";
export { getDeployments } from "./deployments";
export { getServices } from "./services";
export { getJobs } from "./jobs";
