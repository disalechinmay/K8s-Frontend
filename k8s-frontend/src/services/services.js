import { API_LOCATION } from "../configs";
import axios from "axios";

export function getServices(namespace) {
	return new Promise((resolve, reject) => {
		axios
			.get(API_LOCATION + "/servicesx", {
				params: { namespace }
			})
			.then(result => result.data)
			.then(result => {
				if (result.status === "FAILURE")
					reject({
						errorDescription:
							"Something went wrong while retrieving services for '" +
							namespace +
							"' namespace!",
						errorSuggestions: [
							"Make sure the backend service is up and running.",
							"Make sure the endpoint being accessed is valid."
						]
					});
				resolve(result);
			})
			.catch(err =>
				reject({
					errorDescription:
						"Something went wrong while retrieving services for '" +
						namespace +
						"' namespace!",
					errorSuggestions: [
						"Make sure the backend service is up and running.",
						"Make sure the namespace is correct."
					]
				})
			);
	});
}
