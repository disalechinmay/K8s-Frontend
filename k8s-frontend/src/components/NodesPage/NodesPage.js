import React, { Component } from "react";
import SmallLoadingPage from "../common/SmallLoadingPage";
import { getNodes } from "../../services";
import "../../assets/styles/common.css";
import NodeCard from "./NodeCard";

class NodesPage extends Component {
	state = { pageLoading: true, nodesListSet: false, nodesList: [] };

	constructor(props) {
		super(props);

		getNodes().then(result => {
			let newState = { ...this.state };

			newState.pageLoading = false;
			newState.nodesListSet = true;
			newState.nodesList = result.payLoad;

			this.setState(newState);
		});
	}

	render() {
		if (this.state.pageLoading) return <SmallLoadingPage />;

		return (
			<React.Fragment>
				{this.state.nodesListSet &&
					this.state.nodesList.map((nodeInfo, index) => {
						return (
							<NodeCard
								key={index + "_NODE_CARD"}
								nodeInfo={nodeInfo}
								index={index}
								refreshState={() => this.props.refreshState()}
							/>
						);
					})}
			</React.Fragment>
		);
	}
}

export default NodesPage;
