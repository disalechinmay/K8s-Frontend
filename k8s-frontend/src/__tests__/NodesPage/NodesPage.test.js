import { mount } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";
import { NodesPage } from "../../components/NodesPage";
import { getNodes } from "../../services";
import { PAGE_LOAD_LIMIT_TIME } from "../../configs";

jest.mock("../../services");

describe("NodesPage Tests", () => {
	test("Check if SmallLoadingPage renders", () => {
		getNodes.mockImplementationOnce(() => Promise.resolve());

		let wrapper = mount(<NodesPage />);
		expect(wrapper.find("SmallLoadingPage").exists()).toBeTruthy();
	});

	test("Reject getNodes service call.", async () => {
		getNodes.mockImplementationOnce(() => Promise.reject());

		let wrapper = mount(<NodesPage />);
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		expect(wrapper.state("errorSet")).toBeTruthy();
		expect(wrapper.find("SmallErrorPage").exists()).toBeTruthy();
	});

	test("Resolve getNodes service call with non empty payload.", async () => {
		getNodes.mockImplementationOnce(() =>
			Promise.resolve({
				payLoad: [
					{
						nodeAnnotations: {
							"kubeadm.alpha.kubernetes.io/cri-socket":
								"/var/run/dockershim.sock",
							"node.alpha.kubernetes.io/ttl": "0",
							"volumes.kubernetes.io/controller-managed-attach-detach":
								"true"
						},
						nodeCapacity: {
							cpu: "4",
							"ephemeral-storage": "34471692Ki",
							"hugepages-1Gi": "0",
							"hugepages-2Mi": "0",
							memory: "3963180Ki",
							pods: "110"
						},
						nodeLabels: {
							"beta.kubernetes.io/arch": "amd64",
							"beta.kubernetes.io/os": "linux",
							"kubernetes.io/arch": "amd64",
							"kubernetes.io/hostname": "minikube",
							"kubernetes.io/os": "linux",
							"node-role.kubernetes.io/master": ""
						},
						nodeName: "minikube"
					}
				],
				status: "SUCCESS",
				statusDetails: "Returning data from /nodes endpoint."
			})
		);

		let wrapper = mount(<NodesPage />);
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		expect(wrapper.state("errorSet")).toBeFalsy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	test("Resolve getNodes service call with empty payload.", async () => {
		getNodes.mockImplementationOnce(() =>
			Promise.resolve({
				payLoad: [],
				status: "SUCCESS",
				statusDetails: "Returning data from /nodes endpoint."
			})
		);

		let wrapper = mount(<NodesPage />);
		await new Promise(resolve => setTimeout(resolve, PAGE_LOAD_LIMIT_TIME));
		wrapper.update();

		expect(wrapper.state("errorSet")).toBeFalsy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
