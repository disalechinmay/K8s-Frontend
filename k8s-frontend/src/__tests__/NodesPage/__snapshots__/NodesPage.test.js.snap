// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodesPage Tests Resolve getNodes service call with empty payload. 1`] = `<NodesPage />`;

exports[`NodesPage Tests Resolve getNodes service call with non empty payload. 1`] = `
<NodesPage>
  <NodeCard
    key="0_NODE_CARD"
    nodeInfo={
      Object {
        "nodeAnnotations": Object {
          "kubeadm.alpha.kubernetes.io/cri-socket": "/var/run/dockershim.sock",
          "node.alpha.kubernetes.io/ttl": "0",
          "volumes.kubernetes.io/controller-managed-attach-detach": "true",
        },
        "nodeCapacity": Object {
          "cpu": "4",
          "ephemeral-storage": "34471692Ki",
          "hugepages-1Gi": "0",
          "hugepages-2Mi": "0",
          "memory": "3963180Ki",
          "pods": "110",
        },
        "nodeLabels": Object {
          "beta.kubernetes.io/arch": "amd64",
          "beta.kubernetes.io/os": "linux",
          "kubernetes.io/arch": "amd64",
          "kubernetes.io/hostname": "minikube",
          "kubernetes.io/os": "linux",
          "node-role.kubernetes.io/master": "",
        },
        "nodeName": "minikube",
      }
    }
  >
    <div
      className="card flex flex-column"
    >
      <span
        className="title w-100"
      >
        <span
          className="fa fa-lastfm"
        />
        â€ƒ
        minikube
      </span>
      <div
        className="flex flex-row space-around"
      >
        <span>
          <span
            className="big-number"
          >
            4
          </span>
          <br />
          CPU
        </span>
        <span>
          <span
            className="big-number"
          >
            34471692Ki
          </span>
          <br />
          Storage
        </span>
        <span>
          <span
            className="big-number"
          >
            3963180Ki
          </span>
          <br />
          Memory
        </span>
        <span>
          <span
            className="big-number"
          >
            110
          </span>
          <br />
          Pods
        </span>
      </div>
    </div>
  </NodeCard>
</NodesPage>
`;
